pkg/controller/handlers/projectinvitation/projectinvitation.go

```diff
@@ -0,0 +1,56 @@
+package projectinvitation
+
+import (
+	"time"
+
+	"github.com/obot-platform/nah/pkg/router"
+	"github.com/obot-platform/obot/apiclient/types"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func SetRespondedTime(req router.Request, _ router.Response) error {
+	invitation := req.Object.(*v1.ProjectInvitation)
+
+	switch invitation.Spec.Status {
+	case string(types.ProjectInvitationStatusAccepted), string(types.ProjectInvitationStatusRejected):
+		if invitation.Status.RespondedTime.IsZero() {
+			invitation.Status.RespondedTime = &metav1.Time{Time: time.Now()}
+			return req.Client.Status().Update(req.Ctx, invitation)
+		}
+	}
+
+	return nil
+}
+
+// Expiration sets the status of the project invitation to expired if it is pending and has not been accepted or rejected after 7 days.
+func Expiration(req router.Request, resp router.Response) error {
+	invitation := req.Object.(*v1.ProjectInvitation)
+
+	if invitation.Spec.Status != string(types.ProjectInvitationStatusPending) {
+		return nil
+	}
+
+	if time.Since(invitation.CreationTimestamp.Time) > 7*24*time.Hour {
+		invitation.Spec.Status = string(types.ProjectInvitationStatusExpired)
+		invitation.Status.RespondedTime = &metav1.Time{Time: time.Now()}
+		return req.Client.Status().Update(req.Ctx, invitation)
+	}
+	resp.RetryAfter(7*24*time.Hour - time.Since(invitation.CreationTimestamp.Time))
```

Comment:
Only retry if the next time is within 10 hours, otherwise the auto re-enqueue the world will handle it.

Reply:
fixed